server/weaverbird/backends/sql_translator/steps/totals.py

```diff
@@ -0,0 +1,102 @@
+from distutils import log
+from typing import List, Sequence
+
+from weaverbird.backends.sql_translator.steps.utils.query_transformation import (
+    build_selection_query,
+)
+from weaverbird.backends.sql_translator.types import (
+    SQLPipelineTranslator,
+    SQLQuery,
+    SQLQueryDescriber,
+    SQLQueryExecutor,
+    SQLQueryRetriever,
+)
+from weaverbird.pipeline.steps import TotalsStep
+from weaverbird.pipeline.steps.aggregate import Aggregation
+from weaverbird.pipeline.steps.totals import TotalDimension
+from weaverbird.pipeline.types import ColumnName
+from weaverbird.utils.iter import combinations
+
+
+def select_total_dimensions(total_dimensions: List[TotalDimension]) -> List[str]:
+    selects = []
+    for dimension in total_dimensions:
+        selects.append(
+            f"CASE WHEN GROUPING({dimension.total_column}) = 0 THEN {dimension.total_column} ELSE '{dimension.total_rows_label}' END"
+        )
+    return selects
```

Comment:
Since this function is used in only one place, and it's just a loop, is it possible to merge that loop directly inside the function `make_totals_query` at line 46 ? 🤔 
That will prevent to have to write unit tests for this small function that is use only in one place !

Reply:
Same answer as above. It is not because a function exists that it must be unit tested. I'd rather keep this function separated for better readability.

Reply:
ok

Reply:
Good point! The rule would be "if a function is exposed as a public API of the component unit, then it must be uni tested".
@luc-leonard one way to ensure everyone know it's not exposed would be to tag it "private' (even if the concept is just a convention: in python, one would prefix it by an underscore)

https://www.python.org/dev/peps/pep-0008/#designing-for-inheritance
> Public attributes should have no leading underscores.